32blit-stm32/CMakeLists.txt

```diff
@@ -100,19 +100,24 @@ target_compile_definitions(BlitHalSTM32
 		-DSTM32H750xx
 )
 
+target_compile_options(BlitHalSTM32
+	PRIVATE
+		-fpermissive
+)
+
 set(MCU_LINKER_SCRIPT "${CMAKE_CURRENT_LIST_DIR}/${MCU_LINKER_SCRIPT}" PARENT_SCOPE)
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -specs=nano.specs -T ${CMAKE_CURRENT_LIST_DIR}/${MCU_LINKER_SCRIPT}" PARENT_SCOPE)
 
 function(blit_executable NAME SOURCES)
-	add_executable(${NAME}.elf $<TARGET_OBJECTS:BlitHalSTM32> ${SOURCES} ${ARGN})
-  	target_link_libraries(${NAME}.elf BlitEngine)
-	set_target_properties(${NAME}.elf PROPERTIES LINK_DEPENDS ${MCU_LINKER_SCRIPT})
-	set_property(TARGET ${NAME}.elf APPEND_STRING PROPERTY LINK_FLAGS " -Wl,-Map=${NAME}.map,--cref")
-  	add_custom_command(TARGET ${NAME}.elf POST_BUILD
+	add_executable(${NAME} $<TARGET_OBJECTS:BlitHalSTM32> ${SOURCES} ${ARGN})
+	target_link_libraries(${NAME} BlitEngine)
+	set_target_properties(${NAME} PROPERTIES LINK_DEPENDS ${MCU_LINKER_SCRIPT} SUFFIX ".elf")
+	set_property(TARGET ${NAME} APPEND_STRING PROPERTY LINK_FLAGS " -Wl,-Map=${NAME}.map,--cref")
+	add_custom_command(TARGET ${NAME} POST_BUILD
 		COMMENT "Building ${NAME}.bin\nBuilding ${NAME}.dfu"
```

Comment:
+1 for getting rid of the .elf prefix. I was meaning to do this some time. This makes it much easier for user to override settings on the target because it always has the same name.

Reply:
Yep, exactly what I was thinking.